learningequality/kolibri-design-system
# Add KSelect to KDS.

## Description
* Does a minimally modifying port of the KSelect from Kolibri to KDS
* This is almost completely a copy paste + adding to the KThemePlugin, however, some small changes were needed:
  * Import paths from KDS were changed to local imports
  * KSelect used the `commonCoreStrings` mixin in Kolibri, so this is substituted by having a prop to provide text for the clear action.
  * While doing the above, I noticed raw unwrapped text in the component - I replaced this with 'noResultsText` but did not expose it on the KSelect component, because it is currently not used anywhere.

There is almost certainly some refactor that can be done here to avoid the extra component nesting that still exists here, but that seemed better to defer to later for ease of review.

#### Issue addressed
Fixes [#339](https://github.com/learningequality/kolibri-design-system/issues/339)

### Before/after screenshots
Linked KSelect seems to function as before:
![image](https://user-images.githubusercontent.com/1680573/186778889-a717e33d-59d5-4e10-a95c-b51ce86a9997.png)

![image](https://user-images.githubusercontent.com/1680573/186779283-1184f05a-46c6-4a71-90f3-a047bf3d278a.png)


## Steps to test
Link to Kolibri PR https://github.com/learningequality/kolibri/pull/9667

Check instances of KSelect.

The `clearable`/`clearText` props are only used in the resource search panel

## Testing checklist



## Reviewer guidance



## Post-merge updates and tracking


## Comments
